openff/toolkit/topology/_mm_molecule.py

```diff
@@ -494,6 +495,15 @@ def generate_unique_atom_names(self):
 
             atom.name = symbol + str(element_counts[symbol]) + "x"
 
+    @property
+    def has_unique_atom_names(self) -> bool:
+        """``True`` if the molecule has unique atom names, ``False`` otherwise."""
+        # TODO: Let _SimpleAtoms have names?
+        for atom in self.atoms:
+            if not (hasattr(atom, "name")):
+                return False
```

Comment:
We decided they should have names (defaulting to the empty string), so I'm updating this